config/src/main/java/io/scalecube/config/ConfigRegistryImpl.java

```diff
@@ -156,15 +170,38 @@ public int intValue(String name, int defaultValue) {
   }
 
   @Override
+  public DurationConfigProperty durationProperty(String name) {
+    DurationConfigPropertyImpl configProperty = new DurationConfigPropertyImpl();
+    configProperty.configure(name);
+    return configProperty;
+  }
+
+  @Override
+  public Duration durationValue(String name, Duration defaultValue) {
+    return durationProperty(name).get(defaultValue);
+  }
+
+  @Override
+  public <T> ListConfigProperty<T> listProperty(String name, Class<T> elType) {
```

Comment:
Instead of it, lets create separate method for each supported class - like booleanListProperty, integerListProperty etc.